reporter/html.js

```diff
@@ -329,44 +353,106 @@ function toolbarLooseFilter() {
 	}
-	moduleFilterHtml += "</select>";
-
-	return moduleFilterHtml;
+	return dropDownHtml;
 }
 
-function toolbarModuleFilter() {
-	var toolbar = id( "qunit-testrunner-toolbar" ),
+function toolbarModuleFilter () {
+	var dropDownHtml = moduleDropDownHtml(),
+		toolbar = id( "qunit-testrunner-toolbar" ),
+		label = document.createElement( "label" ),
 		moduleFilter = document.createElement( "span" ),
-		moduleFilterHtml = toolbarModuleFilterHtml();
+		moduleSearch = document.createElement( "input" ),
+		dropDownContainer = document.createElement( "div" ),
+		dropDownList = document.createElement( "ul" ),
+		clearFilter = document.createElement( "li" );
 
-	if ( !toolbar || !moduleFilterHtml ) {
+	if ( !toolbar || !dropDownHtml ) {
 		return false;
 	}
 
-	moduleFilter.setAttribute( "id", "qunit-modulefilter-container" );
-	moduleFilter.innerHTML = moduleFilterHtml;
+	moduleSearch.placeholder = "Select modules";
 
-	addEvent( moduleFilter.lastChild, "change", applyUrlParams );
+	label.innerHTML = "Module: ";
+
+	clearFilter.innerHTML = "< All modules ><hr/>";
+
+	clearFilter.onclick = function() {
+		var i,
+			modulesList = dropDownList.getElementsByTagName( "input" );
+		for ( i = 0; i < modulesList.length; i++ )  {
+			modulesList[ i ].checked = false;
+		}
+		applyUrlParams();
+	};
+
+	moduleFilter.id = "qunit-modulefilter-container";
+	dropDownList.id = "qunit-modulefilter-dropdown-list";
+	moduleSearch.id = "qunit-modulefilter-search";
+	dropDownContainer.id = "qunit-modulefilter-dropdown-container";
+
+	dropDownContainer.style.display = "none";
+
+	dropDownList.appendChild( clearFilter );
+	moduleFilter.appendChild( label );
+	moduleFilter.appendChild( moduleSearch );
+	moduleFilter.appendChild( dropDownContainer ) ;
+
+	// Enables show/hide for the dropdown
+	addEvent( moduleSearch, "focus", function() {
+		var dropDownList = id( "qunit-modulefilter-dropdown-container" );
+		if ( dropDownList.style.display === "none" ) {
+			addEvent( document, "click", function hideHandler( e )  {
+				if ( !id( "qunit-modulefilter-container" ).contains( e.target ) ) {
+					id( "qunit-modulefilter-dropdown-container" ).style.display = "none";
+					removeEvent( document, "click", hideHandler );
+				}
+			} );
+			dropDownList.style.display = "block";
+		}
+	} );
+
+	// Enables search to the module filter
+	addEvent( moduleSearch, "input", function() {
+		var i,
+		moduleText,
+		searchText = this.value.toLowerCase(),
+		listItems = id( "qunit-modulefilter-dropdown-list" ).children;
+
+		if ( !searchText ) {
+			for ( i = 0; i < listItems.length; i++ ) {
+				listItems[ i ].style.display = "";
+			}
+		} else {
+			for ( i = 0; i < listItems.length; i++ ) {
+				moduleText = listItems[ i ].textContent.toLowerCase();
+				if ( moduleText.indexOf( searchText ) > -1 ) {
+					listItems[ i ].style.display = "";
+				} else {
+					listItems[ i ].style.display = "none";
+				}
+			}
+		}
+	} );
 
+	dropDownList.insertAdjacentHTML( "beforeend", dropDownHtml );
```

Comment:
It won't work on Android<4.3 or iOS<8.4. Even [jQuery 3.0 supports](https://github.com/jquery/jquery.com/blob/jquery-3/pages/browser-support.html#L18-L22) Android 4.0+ and iOS 7+.